sample-code/src/orchestration.ts

```diff
@@ -1,133 +1,170 @@
+ * Apply a content filter to LLM requests, filtering any hateful input.
  */
-async function orchestrationCompletionFiltering(): Promise<string | undefined> {
+export async function orchestrationInputFiltering(): Promise<void> {
+  // create a filter with minimal thresholds for hate and violence
+  // lower numbers mean more strict filtering
   const filter = buildAzureContentFilter({ Hate: 0, Violence: 0 });
   const orchestrationClient = new OrchestrationClient({
-    llm: {
-      model_name: 'gpt-4-32k',
-      model_params: {}
-    },
-    templating: {
-      template: [{ role: 'user', content: '{{?input}}' }]
-    },
+    llm,
+    templating,
+    // configure the filter to be applied for both input and output
     filtering: {
-      input: filter,
-      output: filter
+      input: filter
     }
   });
 
   try {
-    // Call the orchestration service.
-    const response = await orchestrationClient.chatCompletion({
+    // trigger the input filter, producing a 400 - Bad Request response
+    await orchestrationClient.chatCompletion({
       inputParams: { input: 'I hate you!' }
     });
-    // Access the response content.
-    return response.getContent();
+    throw new Error('Input was not filtered as expected.');
   } catch (error: any) {
-    // Handle the case where the output was filtered.
-    return `Error: ${error.message}`;
+    if (error.response.status === 400) {
+      logger.info('Input was filtered as expected.');
+    } else {
+      throw error;
+    }
   }
 }
 
 /**
- * Ask about the capital of France and send along custom request configuration.
- * @returns The message content from the orchestration service in the generative AI hub.
+ * Apply a content filter to LLM requests, filtering any hateful output.
+ * @returns The orchestration service response.
  */
-async function orchestrationCompletionRequestConfig(): Promise<
-  string | undefined
-> {
+export async function orchestrationOutputFiltering(): Promise<OrchestrationResponse> {
+  // output filters are build in the same way as input filters
+  // set the thresholds to the minimum to maximize the chance the LLM output will be filtered
+  const filter = buildAzureContentFilter({ Hate: 0, Violence: 0 });
   const orchestrationClient = new OrchestrationClient({
-    llm: {
-      model_name: 'gpt-4-32k',
-      model_params: {}
-    },
-    templating: {
-      template: [{ role: 'user', content: 'What is the capital of France?' }]
+    llm,
+    templating,
+    filtering: {
+      output: filter
+    }
+  });
+  // trigger the output filter
+  // note: reliably triggering the output filter is a bit of a challenge
+  // LLMs are fine-tuned to not respond with explicit or hateful content
+  // Instead, they often respond with something like "Sorry, I can't assist with that" when asked to generate explicit content
+  // The following prompt seems to work well with GPT-4o, but may not work with other models
```

Comment:
[req] I would always start a comment with capital letter and end a line with a dot. Same for all other comments.
[pp] I would avoid the term `seems to work` and `may not work`.

Reply:
Changed it to a longer comment block to be consistent with the rest of the commenting style.

For now, I left the `seems to` and `may not` in, to intentionally indicate the uncertainty we are dealing with here. Models change over time and even very slight modifications to the prompt might yield different results.